versioned/storage/dynamodb/src/intTest/java/org/projectnessie/versioned/storage/dynamodb/ITDynamoDBPersist.java

```diff
@@ -15,54 +15,148 @@
 import org.projectnessie.versioned.storage.common.persist.Obj;
 import org.projectnessie.versioned.storage.common.persist.ObjId;
 import org.projectnessie.versioned.storage.common.persist.Persist;
 import org.projectnessie.versioned.storage.commontests.AbstractPersistTests;
+import org.projectnessie.versioned.storage.testextension.BackendTestFactory;
 import org.projectnessie.versioned.storage.testextension.NessieBackend;
 import org.projectnessie.versioned.storage.testextension.NessiePersist;
 import org.projectnessie.versioned.storage.testextension.PersistExtension;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 
 @NessieBackend(DynamoDBBackendTestFactory.class)
-@ExtendWith({PersistExtension.class, SoftAssertionsExtension.class})
 public class ITDynamoDBPersist extends AbstractPersistTests {
-  @InjectSoftAssertions protected SoftAssertions soft;
-
-  @NessiePersist protected Persist persist;
-
-  @Test
-  public void dynamoDbHardItemSizeLimit() throws Exception {
-    persist.storeObj(
-        contentValue(ObjId.randomObjId(), "foo", 42, ByteString.copyFrom(new byte[350 * 1024])));
-
-    persist.storeObjs(
-        new Obj[] {
-          contentValue(ObjId.randomObjId(), "foo", 42, ByteString.copyFrom(new byte[350 * 1024]))
-        });
-
-    // DynamoDB's hard 400k limit
-    soft.assertThatThrownBy(
-            () ->
-                persist.storeObj(
-                    contentValue(
-                        ObjId.randomObjId(), "foo", 42, ByteString.copyFrom(new byte[400 * 1024]))))
-        .isInstanceOf(ObjTooLargeException.class);
-    soft.assertThatThrownBy(
-            () ->
-                persist.storeObjs(
-                    new Obj[] {
+
+  @Nested
+  @ExtendWith({PersistExtension.class, SoftAssertionsExtension.class})
+  public class DynamoDbHardItemSizeLimits {
+    @InjectSoftAssertions protected SoftAssertions soft;
+
+    @NessiePersist protected Persist persist;
+
+    @Test
+    public void dynamoDbHardItemSizeLimit() throws Exception {
+      persist.storeObj(
+          contentValue(ObjId.randomObjId(), "foo", 42, ByteString.copyFrom(new byte[350 * 1024])));
+
+      persist.storeObjs(
+          new Obj[] {
+            contentValue(ObjId.randomObjId(), "foo", 42, ByteString.copyFrom(new byte[350 * 1024]))
+          });
+
+      // DynamoDB's hard 400k limit
+      soft.assertThatThrownBy(
+              () ->
+                  persist.storeObj(
                       contentValue(
-                          ObjId.randomObjId(), "foo", 42, ByteString.copyFrom(new byte[400 * 1024]))
-                    }))
-        .isInstanceOf(ObjTooLargeException.class);
+                          ObjId.randomObjId(),
+                          "foo",
+                          42,
+                          ByteString.copyFrom(new byte[400 * 1024]))))
+          .isInstanceOf(ObjTooLargeException.class);
+      soft.assertThatThrownBy(
```

Comment:
Duplicated code?

Reply:
Potentially, although I did not modify the body of that test at all so the duplication was there already.

Reply:
OK let's leave it, as it's not related to your changes 👍